File: src/02-challenge/solution.js
/**
 * @param {string[]} names
 * @returns {string}
 */
export default function createFrame(names) {
	/** @type {string} */
	let frame

	/** @type {number} */
	let longestName = 0

	for (const name of names) {
		if (name.length > longestName) longestName = name.length
	}

	frame = '*'.repeat(longestName + 4) + '\n'
	for (const name of names) frame += `* ${name.padEnd(longestName, ' ')} *\n`
	frame += '*'.repeat(longestName + 4)

	return frame
}

File: src/02-challenge/solution.py
def createFrame(names: list[str]) -> str:
    longest_name: int = 0

    for name in names:
        longest_name = max(longest_name, len(name))

    frame: str = "*" * (longest_name + 4) + "\n"
    for name in names:
        frame += "* " + name.ljust(longest_name, " ") + " *\n"
    frame += "*" * (longest_name + 4)

    return frame
